src/App.tsx

```diff
@@ -1,19 +1,153 @@
-import React from 'react';
+import { FC, memo, useState } from 'react';
+import 'bulma/css/bulma.min.css';
 import './App.css';
-
+import { PreparedTodos, Todo, User } from './appTypeDefs';
+import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+
+function prepareTodos(
+  todoList: Todo[],
+  usersList: User[],
+): PreparedTodos[] {
+  return todoList.map(todo => ({
+    ...todo,
+    user: usersList.find(user => user.id === todo.userId),
+  }));
+}
+
+export const preparedTodos: PreparedTodos[] = prepareTodos(todos, users);
+
+const App: FC = () => {
+  const [todoTitle, setTitle] = useState('');
+  const [userId, setUserId] = useState('');
+  const [toDoList, setToDoList] = useState([...preparedTodos]);
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+  let uName = '';
+
+  const validateInput = () => {
+    if (!userId) {
+      setUserError('Please choose a user');
+    }
+
+    if (!todoTitle) {
+      setTitleError('Please enter the title');
+    }
+  };
+
+  const getUser = () => users.find(user => user.name === userId);
+
+  const addToDo = () => {
+    if (todoTitle && userId) {
+      setToDoList(prevState => ([
+        ...prevState,
+
+        {
+          userId: getUser()?.id,
+          id: prevState.length + 1,
+          title: todoTitle,
+          completed: false,
+          user: getUser(),
+        },
+      ]));
+
+      setTitle('');
+    } else {
+      validateInput();
+    }
+  };
 
-const App: React.FC = () => {
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
+    <>
+      <div className="App container">
+        <h1 className="title is-1">To-Do List</h1>
+
+        <form
+          action="#/"
+          method="POST"
+          className="Form"
+          onSubmit={(e) => {
+            e.preventDefault();
```

Comment:
i think you should use form submit handler and use prevent default there

Reply:
Okay thanx done